src/routes/game/components/dialogs/components/ScrapDialog.vue

```diff
@@ -51,33 +97,162 @@
   </BaseDialog>
 </template>
 
-<script>
+<script setup>
+import { ref, computed, useTemplateRef } from 'vue';
+import { useDisplay } from 'vuetify';
 import { useI18n } from 'vue-i18n';
+import { onLongPress } from '@vueuse/core';
+import { useGameStore } from '_/src/stores/game';
 import CardListSortable from '@/routes/game/components/CardListSortable.vue';
 import BaseDialog from '@/components/BaseDialog.vue';
+import GameCard from '@/routes/game/components/GameCard.vue';
 
-export default {
-  name: 'ScrapDialog',
-  components: {
-    CardListSortable,
-    BaseDialog,
-  },
-  props: {
-    scrap: {
-      type: Array,
-      required: true,
-    },
-  },
-  setup() {
-    const { t } = useI18n();
-    return { t };
-  },
-  data() {
-    return {
-      show: false,
-    };
+const props = defineProps({
+  scrap: {
+    type: Array,
+    required: true,
   },
+});
+
+const { t } = useI18n();
+const gameStore = useGameStore();
+
+const show = ref(false);
+const lastStraightenedCardId = ref(null);
+const scrapWrapper = useTemplateRef('scrap');
+const isLongPressing = ref(false);
+
+const scrapDisplay = computed(() => props.scrap.slice(-10));
+const straightendIndex = computed(() => scrapDisplay.value.map(card => card.id).indexOf(lastStraightenedCardId.value));
+const threesTransition = computed(() => gameStore.lastEventPlayerChoosing ? `threes-player` : `threes-opponent`);
+const { xs } = useDisplay();
+
+// Straighten pile on long press; prevent opening dialog
+onLongPress(scrapWrapper, () => {
+  isLongPressing.value = true;
+  const topScrapCardId = props.scrap[props.scrap.length -1].id;
+  if (lastStraightenedCardId.value === topScrapCardId) {
+    lastStraightenedCardId.value = null;
+  } else {
+    lastStraightenedCardId.value = topScrapCardId;
+  }
+}, {
+  stop: true
+});
+
+// Prevent dialog on long press
+function onActivatorClick(e) {
+  if (isLongPressing.value) {
+    e.preventDefault();
+    e.stopPropagation();
+    isLongPressing.value = false;
+    return;
+  }
+  // Allow normal click
 };
```

Comment:
Question: should I group these by feature rather than data type? What would that look like?

Offhand, there's:

- Dialog activation

- tidying the pile